### ISSUE TITLE
Should we zoom out when changing screens?

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/joist#682]
In phetsims/ratio-and-proportion#282 we are going to move focus from the NavigationBarScreenButton to the screen itself so that focus and the virtual cursor starts at the top of the screen. Our pan and zoom feature will try to pan to the center of the focused node, clicking a NavigationBarScreenButton while zoomed in will pan to the center of the zoomed in screen.

We wanted to make sure this was OK behavior.  A potential improvement could be to zoom out when changing screens, but we weren't sure if that was correct either. @terracoda do you have any thoughts on how it should behave?

**Comment by terracoda** [comment]
@jessegreenberg, I think this a great question.
After zooming in and switching screens in RaP, I think your instinct is correct.

We should zoom out when switching screens.
This would allow for some orientation to occur on a new screen before zooming in again to interaction or to take a closer look.

Some some screens are quite different, so I feel like zooming out on a screen change makes sense.

**Comment by jessegreenberg** [comment]
Done in the above commit. @zepumph I am curious if this will have an impact on the feature with PhET-iO? Like if the screenProperty is set on startup in the downstream sim which resets the transform, it seems it will be impossible to load a sim with a predetermined pan/zoom. That may be acceptable for now, but want to know your thoughts.

**Comment by zepumph** [comment]
In general this is how we solve this. Basically saying that state setting should not change this.


```diff
Index: js/Sim.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/Sim.js b/js/Sim.js
--- a/js/Sim.js	(revision dfcb8cf5ae3127dc67a56c36f0e1e421ab428825)
+++ b/js/Sim.js	(date 1612223936683)
@@ -792,9 +792,12 @@
       currentScreen.view.focusable = true;
       currentScreen.view.focus();
 
-      // zoom out again after changing screens so we don't pan to the center of the focused SCreenView,
-      // and so user has an overview of the new screen, see https://github.com/phetsims/joist/issues/682
-      animatedPanZoomSingleton.listener.resetTransform();
+      if ( !this.isSettingPhetioStateProperty.value ) {
+
+        // zoom out again after changing screens so we don't pan to the center of the focused SCreenView,
+        // and so user has an overview of the new screen, see https://github.com/phetsims/joist/issues/682
+        animatedPanZoomSingleton.listener.resetTransform();
+      }
     } );
 
     // layer for popups, dialogs, and their backgrounds and barriers

```

This doesn't change behavior like when the screenProperty changes from studio, but I think we can keep that in for now, and see if we get complaints. What do you think? Will you apply this patch and try it out in the state wrapper?

**Comment by jessegreenberg** [comment]
Thanks! That is exactly what I was looking for. I added that line and tested in state wrapper by first setting "Set State Rate" slider to zero, going to a different screen in upstream sim and changing zoom level, then pressing "Set State Now". I saw that zoom level was carried to downstream sim in the state wrapper. I also saw it working as I would expect in studio, zoom level was carried over to new sim after pressing "Preview Sim".  Neither of these worked before adding this line.

**Comment by jessegreenberg** [comment]
I think this is ready for review, @terracoda can you please try this change out in a multi-screen sim? The zoom level should reset ever time you change screens. If it is working as you expect, feel free to close.

**Comment by terracoda** [comment]
This work great on my Mac with Safari.

**Comment by terracoda** [comment]
Pan and Zoom doesn't work at all with VoiceOver enabled. I think that is intended.
And with VO not enabled, the Pan and Zoom between screen seems to work fine, i.e., the screen zooms back out to give the learner a view of the entire sim screen.

I think we can close this issue.

**Comment by zepumph** [comment]
Note that I cherry-picked this into the RAP 1.0 rc branch.
